src/cogito/package.d

```diff
@@ -36,6 +36,30 @@ private Result runOnFile(string file)
     return typeof(return)(visitor.source);
 }
 
+/**
+ * Map a filename to an array of files.
+ *
+ * Params:
+ *     filename = File path.
+ *
+ * Returns: Array of filenames.
+ *   If the filename points to a folder all d-files are returned,
+ *   If the filename points to a file an array with only that file is returned.
+ */
+string[] toFiles(string filename)
+{
+    import std.file;
+    if (filename.isFile)
+    {
+        return [filename];
+    }
+    if (filename.isDir)
+    {
+        return filename.dirEntries("*.d", SpanMode.breadth).map!("a.name").array;
```

Comment:
When I'm mean and create a directory `test.d`, I get a very misleading error message:
```
 read error: Is a directory
Specify path to file 'src/test.d' with -I switch
```

Reply:
In my optinion it's not related. This error already exists if you pass a non-existing file to the DMD frontend. It should be fixed at another place, probalby in `runOnFile()`.

Reply:
i can prepare a patch that just filters such files out ... if you give an non existing file, its kind of your fault, if the directory traversal picks up a directory and provides that as a file, its a bug. ...

Reply:
I think not-existing files shouldn't be just ignored. Directories should be handled as directories and files as files. If a file or directory doesn't exist, and the user doesn't get an error message, he may think that the score is fine, but in the reality he just misspelled the path. Maybe it shouldn't be a hard error and cogito can continue, but at least a warning “File or directory bla-bla doesn't exist” would be useful.

Reply:
yes ... this would be meaningful. i think even a hard error is good! because if you specify a file or directory that does not exist (it e.g. might have existed previously, but was refactored away and the cogito config forgotten). but this does not apply to folders that end with .d and are picked up during directory traversal.

Reply:
Ah, you mean existing directories with `.d` suffix, yes, I agree, they can be filtered out.